drivers/overlay/peerdb.go

```diff
@@ -244,13 +237,12 @@ func (d *driver) peerDbUpdateSandbox(nid string) {
 
 		peerOps = append(peerOps, op)
 	}
-	pMap.Unlock()
 
 	for _, op := range peerOps {
 		op()
 	}
+	pMap.Unlock()
```

Comment:
Thats the part I am looking at as well. Usage of the WaitGroup needs to be fixed. But its better to avoid taking a lock around peerAdd which is a pretty expensive call, with multiple netlink calls to the kernel.